### ISSUE TITLE
Works With Google Pixel 9 Pro!

### ISSUE THREAD

**Issue by Danc2050** [sasodoma/uwb-ranging#2]
I was able to get this working with a Google Pixel 9 Pro for distance measurements! No AoA...perhaps more devices are needed for that.

**Comment by sasodoma** [comment]
That's great. Did you also try swapping roles (controller, controlee)? For me it only worked with the phone being the controlee.

**Comment by Danc2050** [comment]
> That's great. Did you also try swapping roles (controller, controlee)? For me it only worked with the phone being the controlee.

I just tried this and yes it works! And it actually works better than the Controller mode as I get Azimuth and Elevation measurements in addition to distance...and distance seems measurements seem more accurate too!

I am assuming this means that the Google Pixel 9 Pro has more advanced UWB hardware or software capabilities than Qorvo, which allows it to provide richer data (such as Azimuth and Elevation) when acting as a Controlee compared to when it acts as a Controller...but I am new to this and still learning.

Unless I can find a better hardware, I will try expanding this idea to work with more than one device.

**Comment by sasodoma** [comment]
I have previously used the Android app from this repository to do ranging between two Android devices (Pixel 8 Pro and a Samsung Galaxy S22 Ultra) so you could also try testing that if you have two UWB capable phones.

**Comment by Danc2050** [comment]
> I have previously used the Android app from this repository to do ranging between two Android devices (Pixel 8 Pro and a Samsung Galaxy S22 Ultra) so you could also try testing that if you have two UWB capable phones.

I assume you're referring to this repo: 
https://github.com/android/connectivity-samples/tree/main/UwbRanging. I have read on a Qorvo forum post that someone got two DWM3001CDK's to interact with that code. 

For my use case, I am most interested in having the DWM3001CDK talk to UWB-enabled phones. Or possibly each other, but ultimately relay the info to the phone.

I just found out they recently uploaded recent development kits [DW3xxx & QM3xxx SDK v1.0.2](https://www.qorvo.com/products/d/da008582) and [Qorvo_Nearby_Interaction](https://www.qorvo.com/products/d/da008212) as can be seen on the [Evaluation Tools page](https://www.qorvo.com/products/p/DWM3001CDK#evaluation-tools). 

I will have to use BLE to interact with the DWM3001CDK board (I am told it has Bluetooth) to get config details. Hopefully I can draw some inspiration from this code and apply it to an Android environment.

**Comment by sasodoma** [comment]
Yes I made the code in this repo specifically with the purpose of simply testing ranging without the need to mess with BLE. I hope it can help you with your application as well.

**Comment by Danc2050** [comment]
Thank you again for all your help in providing this initial work. I'll let you know how far I get with things if you are interested. 

Also, if you'd like to stay in touch via some other method, feel free to email me at connellyd2050 [at] gmail dot com.

**Comment by sasodoma** [comment]
Personally, I only bought the board to play around with for a bit, and currently I've lost the interest to do stuff with it. But I'd be happy to hear what you manage to do with it. While this is technically an issue thread, it's no problem if you use it to post any updates. This way others who find this repo can also read it.

**Comment by Danc2050** [comment]
Also tested out on the updated firmware you uploaded and it works on Google Pixel 9 pro.

**Comment by sasodoma** [comment]
Thanks for the update! Glad to hear that the new script is also working, there were some issues with Python versions, but they should be fixed now.
